gatling-http/src/main/scala/io/gatling/http/Predef.scala

```diff
@@ -15,8 +15,32 @@
  */
 package io.gatling.http
 
+import io.gatling.http.check.HttpCheck
+import io.gatling.http.response.Response
+import io.gatling.http.check.HttpCheckScope
+
+import io.gatling.commons.validation.Success
+
+import io.gatling.core.session._
+import io.gatling.core.check.Check
+import io.gatling.core.check.ConditionalCheck._
+
 object Predef extends HttpDsl {
 
   type Request = org.asynchttpclient.Request
   type Response = io.gatling.http.response.Response
+
+  implicit object HttpCheckWrapper extends CheckWrapper[Response, HttpCheck] {
+    def wrap(check: Check[Response]): HttpCheck = new HttpCheck(check, HttpCheckScope.Body, None)
```

Comment:
That's an issue there. You can't assume scope is Body and bodyUsage is None: it depends on the check.
I suggest:
* we change HttpCheck so scope and bodyUsage are multivalues (Set)
* CheckWrapper can only wrap one given type of Check, so we have a `HttpConditionalCheckWrapper` that can access the inner checks and compute a sum of their scopes and bodyUsages